Readline-Release: 7.0
Patch-ID: readline70-005

Bug-Reported-by:        Nuzhna Pomoshch <nuzhna_pomoshch@yahoo.com>
Bug-Reference-ID:       <1317167476.1492079.1495999776464@mail.yahoo.com>
Bug-Reference-URL:      https://lists.gnu.org/archive/html/bug-readline/2017-05/msg00005.html

Bug-Description:

There are cases where a failing readline command (e.g., delete-char at the end
of a line) can cause a multi-character key sequence to `back up' and attempt
to re-read some of the characters in the sequence.
Index: b/patchlevel
===================================================================
--- a/patchlevel
+++ b/patchlevel
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-4
+5
Index: b/readline.c
===================================================================
--- a/readline.c
+++ b/readline.c
@@ -1057,7 +1057,7 @@ _rl_subseq_result (r, map, key, got_subs
 	/* We probably shadowed a keymap, so keep going. */
 	r = _rl_dispatch (ANYOTHERKEY, m);
     }
-  else if (r && map[ANYOTHERKEY].function)
+  else if (r < 0 && map[ANYOTHERKEY].function)
     {
       /* We didn't match (r is probably -1), so return something to
 	 tell the caller that it should try ANYOTHERKEY for an
@@ -1069,7 +1069,7 @@ _rl_subseq_result (r, map, key, got_subs
       _rl_dispatching_keymap = map;
       return -2;
     }
-  else if (r && got_subseq)
+  else if (r < 0 && got_subseq)		/* XXX */
     {
       /* OK, back up the chain. */
       if (RL_ISSTATE (RL_STATE_MACROINPUT))
